firmware/src/networking/wifi/esp.rs

```diff
@@ -24,11 +29,27 @@ pub async fn network_task(signals: &Signals) {
 		.await
 		.expect("Couldn't connect to wifi");
 
-	let mut network = Network::new(wifi, current_millis);
+	let network = Network::new(wifi, current_millis);
+
+	// Wait till DHCP assigns us an IP
+	let client_ip = loop {
+		network.poll_dhcp().expect("Failed to drive DHCP socket");
```

Comment:
`Network::work` already polls the DCHP server

Reply:
Does that mean that the expect can never happen? In that case we should do something like `.expect("This should be impossible, because Network::work already polled)`

Reply:
Oh I see, you removed it from the code now. resolving.